### PR TITLE
Improve callbacks return types

### PR DESCRIPTION
**Pull request by varsill**

### CODE REVIEW

File: lib/membrane/bin/action.ex [hunk 1]
```diff
@@ -13,9 +13,10 @@ defmodule Membrane.Bin.Action do
   @typedoc """
   Action that manages the end of the component setup.
 
-  By default, component setup ends with the end of `c:Membrane.Bin.handle_setup/2` callback.
+  By default, component setup ends with the end of  callback.
```
Comment (mat-hek) [inline lib/membrane/bin/action.ex:16]: ?

File: lib/membrane/element/action.ex [hunk 2]
```diff
@@ -245,27 +245,31 @@ defmodule Membrane.Element.Action do
...
           | pause_auto_demand
           | resume_auto_demand
-          | forward
-          | start_timer
-          | timer_interval
-          | stop_timer
-          | latency
           | end_of_stream
-          | terminate
+          | redemand
```
Comment (mat-hek) [inline lib/membrane/element/action.ex:265]: this is duplicated

File: lib/membrane/element/action.ex [hunk 3]
```diff
@@ -245,27 +245,31 @@ defmodule Membrane.Element.Action do
...
+
+  @typedoc """
+  Type that defines a union of actions that may be returned from some element callbacks.
+
+  Depending on element type, callback, current playback and other
+  circumstances there may be different actions available.
+  """
+
```
Comment (mat-hek) [inline lib/membrane/element/action.ex:273]:
```suggestion
```

File: lib/membrane/element/action.ex [hunk 4]
```diff
@@ -245,27 +245,31 @@ defmodule Membrane.Element.Action do
...
-          | start_timer
-          | timer_interval
-          | stop_timer
-          | latency
           | end_of_stream
-          | terminate
+          | redemand
+
+  @typedoc """
+  Type that defines a union of actions that may be returned from some element callbacks.
```
Comment (mat-hek) [inline lib/membrane/element/action.ex:268]:
```suggestion
  Type that defines a union of actions that may be returned from most of the element callbacks.
```

File: lib/membrane/element/action.ex [hunk 5]
```diff
@@ -245,27 +245,31 @@ defmodule Membrane.Element.Action do
...
+
+  Depending on element type, callback, current playback and other
+  circumstances there may be different actions available.
```
Comment (mat-hek) [inline lib/membrane/element/action.ex:271]:
```suggestion
  circumstances there may be different actions available.
  Check the typespec and documentation of particular callbacks
  for details.
```

File: lib/membrane/element/base.ex [hunk 6]
```diff
@@ -33,10 +33,10 @@ defmodule Membrane.Element.Base do
...
   alias Membrane.Element.{Action, CallbackContext}
 
-  @typedoc """
-  Type that defines all valid return values from most callbacks.
-  """
-  @type callback_return :: {[Action.t()], Element.state()}
+  # @typedoc """
+  # Type that defines all valid return values from most callbacks.
+  # """
+  # @type callback_return :: {[Action.t()], Element.state()}
```
Comment (mat-hek) [inline lib/membrane/element/base.ex:40]:
```suggestion
```

File: lib/membrane/bin/action.ex (hunk 1 above)
Comment (varsill) [reply lib/membrane/bin/action.ex:16]: done

File: lib/membrane/element/action.ex (hunk 3 above)
Comment (varsill) [reply lib/membrane/element/action.ex:273]: done

File: lib/membrane/element/action.ex (hunk 4 above)
Comment (varsill) [reply lib/membrane/element/action.ex:268]: done

File: lib/membrane/element/action.ex (hunk 5 above)
Comment (varsill) [reply lib/membrane/element/action.ex:271]: done

File: lib/membrane/element/base.ex (hunk 6 above)
Comment (varsill) [reply lib/membrane/element/base.ex:40]: done

**Review by mat-hek — APPROVED** [review]

**Pull request merged**